src/utilities/multiprecision.h

```diff
@@ -23,68 +23,126 @@
 #define LDBL_SUFFIX long_dbl
 
 /*--------------------------------------------------------------------------
-* For Multi-precision build. Only set when hypre 
-* is built with mixed-precision
+* For mixed-precision build only
 *---------------------------------------------------------------------------*/
+
 #if defined(HYPRE_MIXED_PRECISION)
-/* matrix/ solver precision options */
+
+/* object precision options */
 typedef enum 
 {
    HYPRE_REAL_SINGLE,
    HYPRE_REAL_DOUBLE,
    HYPRE_REAL_LONGDOUBLE
+
 } HYPRE_Precision;
-/*--------------------------------------------------------------------------
-* Reset build types for Multi-precision build
-*---------------------------------------------------------------------------*/
+
+/* Set build options */
 #if defined(MP_BUILD_SINGLE)
+
+#define HYPRE_MULTIPRECISION_FUNC(a) CONCAT_(a, FLT_SUFFIX)
+#define HYPRE_ZZZZZPRECISION_FUNC(a) CONCAT_(a, FLT_SUFFIX)  /* RDF: Temporary */
+#define HYPRE_FIXEDPRECISION_FUNC(a) CONCAT_(a, FLT_SUFFIX)
 #define HYPRE_OBJECT_PRECISION HYPRE_REAL_SINGLE
 #define BUILD_MP_FUNC 1
-#undef HYPRE_LONG_DOUBLE
 #ifndef HYPRE_SINGLE
 #define HYPRE_SINGLE 1
 #endif
+#undef  HYPRE_LONG_DOUBLE
+
+#elif defined(MP_BUILD_DOUBLE)
+
+#define HYPRE_MULTIPRECISION_FUNC(a) CONCAT_(a, DBL_SUFFIX)
+#define HYPRE_ZZZZZPRECISION_FUNC(a) CONCAT_(a, DBL_SUFFIX)  /* RDF: Temporary */
+#define HYPRE_FIXEDPRECISION_FUNC(a) CONCAT_(a, DBL_SUFFIX)
+#define HYPRE_OBJECT_PRECISION HYPRE_REAL_DOUBLE
+#define BUILD_MP_FUNC 1
+#undef  HYPRE_SINGLE
+#undef  HYPRE_LONG_DOUBLE
+
 #elif defined(MP_BUILD_LONGDOUBLE)
+
+#define HYPRE_MULTIPRECISION_FUNC(a) CONCAT_(a, LDBL_SUFFIX)
+#define HYPRE_ZZZZZPRECISION_FUNC(a) CONCAT_(a, LDBL_SUFFIX)  /* RDF: Temporary */
+#define HYPRE_FIXEDPRECISION_FUNC(a) CONCAT_(a, LDBL_SUFFIX)
 #define HYPRE_OBJECT_PRECISION HYPRE_REAL_LONGDOUBLE
 #define BUILD_MP_FUNC 1
-#undef HYPRE_SINGLE
+#undef  HYPRE_SINGLE
 #ifndef HYPRE_LONG_DOUBLE
 #define HYPRE_LONG_DOUBLE 1
 #endif
-#elif defined(MP_BUILD_DOUBLE)
-#define HYPRE_OBJECT_PRECISION HYPRE_REAL_DOUBLE
+
+#elif defined(MP_BUILD_DEFAULT)
+
+#define HYPRE_MULTIPRECISION_FUNC(a) a
+#define HYPRE_ZZZZZPRECISION_FUNC(a) CONCAT_(a, def)  /* RDF: Temporary */
```

Comment:
What will this macro return now, in the case when we get rid of Foo_def? Could it be a no-op as before? I suppose this could have implications on the way the name changes are handled in the header files (eg. HYPRE_Krylov.h)?